packages/wrangler/src/hyperdrive/update.ts

```diff
@@ -39,52 +41,92 @@ export function options(yargs: CommonYargsArgv) {
+				describe:
+					"Specifies max duration for which items should persist in the cache, cannot be set when caching is disabled",
+			},
+			swr: {
+				type: "number",
 				describe:
-					"Whether caching query results is disabled for this Hyperdrive config",
+					"Indicates the number of seconds cache may serve the response after it becomes stale, cannot be set when caching is disabled",
 			},
 		})
 		.epilogue(hyperdriveBetaWarning);
 }
 
+const requiredOriginOptions = [
+	"origin-host",
+	"origin-port",
+	"database",
+	"origin-user",
+	"origin-password",
+];
+
+function isOptionSet<T extends object>(args: T, key: keyof T): boolean {
+	return key in args && args[key] !== undefined;
+}
+
 export async function handler(
 	args: StrictYargsOptionsToInterface<typeof options>
 ) {
+	// check if all or none of the required origin fields are set, since we don't allow partial updates of the origin
+	const allOriginFieldsSet = requiredOriginOptions.every((field) =>
+		isOptionSet(args, field as keyof typeof options)
+	);
+	const noOriginFieldSet = requiredOriginOptions.every(
+		(field) => !isOptionSet(args, field as keyof typeof options)
+	);
+
+	if (!allOriginFieldsSet && !noOriginFieldSet) {
+		throw new Error(
+			`When updating the origin, all of the following must be set: ${requiredOriginOptions.join(
+				", "
+			)}`
+		);
+	}
+
 	const config = readConfig(args.config, args);
 
 	logger.log(`🚧 Updating '${args.id}'`);
-	const database = (await getConfig(
-		config,
-		args.id
-	)) as CreateUpdateHyperdriveBody;
-	if (args.originHost) {
-		database.origin.host = args.originHost;
-	}
-	if (args.originPort) {
-		database.origin.port = args.originPort;
-	}
-	if (args.originScheme) {
-		database.origin.scheme = args.originScheme;
-	} else if (!database.origin.scheme) {
-		database.origin.scheme = "postgresql";
-	}
-	if (args.database) {
-		database.origin.database = args.database;
-	}
-	if (args.originUser) {
-		database.origin.user = args.originUser;
+
+	const database: PatchHyperdriveBody = {};
+
+	if (args.name !== undefined) {
+		database.name = args.name;
 	}
-	if (args.originPassword) {
-		database.origin.password = args.originPassword;
+
+	if (allOriginFieldsSet) {
+		database.origin = {
+			host: args.originHost,
+			port: args.originPort,
+			database: args.database,
+			user: args.originUser,
+			password: args.originPassword,
+		};
+		if (args.originScheme !== undefined) {
+			database.origin.scheme = args.originScheme;
+		} else {
+			database.origin.scheme = "postgresql"; // setting default if not passed
+		}
 	}
-	if (args.cachingDisabled !== undefined) {
-		database.caching.disabled = args.cachingDisabled;
+
+	if (args.cachingDisabled || args.maxAge || args.swr) {
+		database.caching = {
+			disabled: args.cachingDisabled,
+			maxAge: args.maxAge,
+			staleWhileRevalidate: args.swr,
+		};
 	}
-	const updated = await updateConfig(config, args.id, database);
+
+	const updated = await patchConfig(config, args.id, database);
```

Comment:
- [ ] This relies on new `PATCH` handler that is not yet available in production. Do not merge until this handler has been released.

Reply:
The PATCH handler has been released and is documented here: https://developers.cloudflare.com/api/operations/patch-hyperdrive